packages/editor/src/components/global-styles-provider/index.js

```diff
@@ -46,38 +46,61 @@ export function mergeBaseAndUserConfigs( base, user ) {
 function useGlobalStylesUserConfig() {
 	const { globalStylesId, isReady, settings, styles, _links } = useSelect(
 		( select ) => {
-			const { getEditedEntityRecord, hasFinishedResolution, canUser } =
-				select( coreStore );
+			const {
+				getEntityRecord,
+				getEditedEntityRecord,
+				hasFinishedResolution,
+				canUser,
+			} = select( coreStore );
 			const _globalStylesId =
 				select( coreStore ).__experimentalGetCurrentGlobalStylesId();
 
-			const record =
-				_globalStylesId &&
-				canUser( 'read', {
-					kind: 'root',
-					name: 'globalStyles',
-					id: _globalStylesId,
-				} )
-					? getEditedEntityRecord(
-							'root',
-							'globalStyles',
-							_globalStylesId
-					  )
-					: undefined;
+			let record;
+			const userCanEditGlobalStyles = canUser( 'update', {
+				kind: 'root',
+				name: 'globalStyles',
+				id: _globalStylesId,
+			} );
+
+			if ( _globalStylesId ) {
+				if ( userCanEditGlobalStyles ) {
+					record = getEditedEntityRecord(
+						'root',
+						'globalStyles',
+						_globalStylesId
+					);
+				} else {
+					record = getEntityRecord(
+						'root',
+						'globalStyles',
+						_globalStylesId,
+						{ context: 'view' }
+					);
+				}
+			}
```

Comment:
Looking at the [item schema definition](https://github.com/WordPress/gutenberg/blob/f073eb1f9f3b900ca347c7c8132fc7646f1bfaea/lib/class-wp-rest-global-styles-controller-gutenberg.php#L532-L573), I see that every property is exposed in `view` context.

I've not tested it, but this conditional call to selectors might not be needed. We should be able to call `getEditedEntityRecord` with `content: view` unless we're planning to review the schema and context for each property.

Reply:
Isn't `edit` the default context in the editor thanks to the entity base URL params?

https://github.com/WordPress/gutenberg/blob/bdfd4696797f19f844057b88664dd48ba2155de7/packages/core-data/src/entities.js#L185-L185

If not, great! 🎉 

Otherwise, I did it the "longhand" way with "view" for now to avoid having to backtracking and adding `edit` in the appropriate places. Was planning on doing that in a follow up.

Reply:
Yes, `edit` is the default context.

I forgot that `getEditedEntityRecord` doesn't accept the `query` parameter :/